articles/creating-a-simple-real-time-chat-app-with-react-socket-io-with-e2e-encryption/index.md

```diff
@@ -0,0 +1,832 @@
+  join_User,
+  get_Current_User,
+  user_Disconnect,
+};
+```
+
+In the above code snippet, the below functions that deals with the user has been created:
+
+The `join_User()` function adds the user to the array of users already declared in the above code. It consists of three keys *id, a username*, and a *room name*, where the room name tells the user the room or group belongs.
+
+The `get_Current_User()` function will take the id of the particular user and return its user object.
+
+In the `user_Disconnect()` function, if a user disconnects or leaves the chat, the function accepts a *user id* and deletes the user object from the array users.
+
+**Step 3**
+
+In this stage, we'll code the file `server.js` as below:
+
+```JavaScript
+const express = require("express");
+const app = express();
+const socket = require("socket.io");
+const color = require("colors");
+const cors = require("cors");
+const { get_Current_User, user_Disconnect, join_User } = require("./dummyuser");
+
+app.use(express());
+
+const port = 8000;
+
+app.use(cors());
+
+var server = app.listen(
+  port,
+  console.log(
+    `Server is running on the port no: ${(port)} `
+      .green
+  )
+);
+
+const io = socket(server);
+
+//initializing the socket io connection 
+io.on("connection", (socket) => {
+  //for a new user joining the room
+  socket.on("joinRoom", ({ username, roomname }) => {
+    //* create user
+    const p_user = join_User(socket.id, username, roomname);
+    console.log(socket.id, "=id");
+    socket.join(p_user.room);
+
+    //display a welcome message to the user who have joined a room
+    socket.emit("message", {
+      userId: p_user.id,
+      username: p_user.username,
+      text: `Welcome ${p_user.username}`,
+    });
+
+    //displays a joined room message to all other room users except that particular user
+    socket.broadcast.to(p_user.room).emit("message", {
+      userId: p_user.id,
+      username: p_user.username,
+      text: `${p_user.username} has joined the chat`,
+    });
+  });
+
+  //user sending message
+  socket.on("chat", (text) => {
+    //gets the room user and the message sent
+    const p_user = get_Current_User(socket.id);
+
+    io.to(p_user.room).emit("message", {
+      userId: p_user.id,
+      username: p_user.username,
+      text: text,
+    });
+  });
+
+  //when the user exits the room
+  socket.on("disconnect", () => {
+    //the user is deleted from array of users and a left room message displayed
+    const p_user = user_Disconnect(socket.id);
+
+    if (p_user) {
+      io.to(p_user.room).emit("message", {
+        userId: p_user.id,
+        username: p_user.username,
+        text: `${p_user.username} has left the chat`,
+      });
+    }
+  });
+});
+```
+
+In the above `server.js` code, we start by importing the modules and functions from the file `dummyuser.js`. The code listens on port *8000* and initializes the socket.
+
+After initializing the socket, the two functions listed below will be called.
```

Comment:
Technically, the function you pass to these event listeners will be called. Please mention the event listeners.